### DISCUSSION TITLE
How do I know where's "inside" and "outside" when using LineZone?

### DISCUSSION THREAD

**Discussion by tiago-hansen** [roboflow/supervision#778] [Q&A] [answered]
Hi everyone!

When I have a vertical LineZone, for example, and one object crosses the line from the left to the right. Does it sum up in "crossed_in" or "crossed_out", and how should I figure it out where is "inside" and "outside" on the image?

And what if I had an inclined or horizontal line, what is the parameter used in the trigger() method of LineZone to differentiate if it should add in "crossed_in" or "crossed_out" attributes?

I'm having a little bit of trouble trying to figure that on my own and would appreciate any help.

Thanks!

**Comment by SkalskiP** [comment]
I/O is just a convention. Please take a look at the drawing below. If you want to reverse the count direction, pass the "end" point as "start" and "start" as 'end". 

![in_and_out_convention](https://github.com/roboflow/supervision/assets/26109316/b7e31121-8a78-4c45-8cdd-423c6bd84638)

**Reply by tiago-hansen** [reply]
Thank you for your answer!

In this case, is there a way I could previously tell, when setting a start and end point, the I/O direction?
Or would I need to set it first at "random", test when an object passes, check if it is the way I want, and if it is not, invert the start and end point?

**Reply by tiago-hansen** [reply]
I can understand it is a convention, but I was not able to understand what the convention is, yet.

**Reply by SkalskiP** [reply, marked as answer]
The convention is on the drawing; it is not random. If you look from the "start" point towards the "end" point, and the box travels from your right to left, it goes out. If from left to right, it is going in. Look at the arrow directions - you will see the pattern.

**Reply by tiago-hansen** [reply]
Perfect! I understand it now.

Thank you!

**Reply by SkalskiP** [reply]
We plan to update LineZone API and create docs. I'll make sure to include this topic there as well.
